data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java

```diff
@@ -138,11 +138,10 @@ public void processStream(final StreamPartition streamPartition) {
                             LOG.debug("Write to buffer for line " + (recordCount - defaultFlushBatchSize) + " to " + recordCount);
                             recordBufferWriter.writeToBuffer(acknowledgementSet, records);
                             records.clear();
-                            partitionCheckpoint.checkpoint(checkPointToken, recordCount);
                             LOG.debug("Perform regular checkpointing for stream Loader");
                             partitionCheckpoint.checkpoint(checkPointToken, recordCount);
+                            successItemsCounter.increment();
```

Comment:
I think we need to increment by `defaultFlushBatchSize` here, but you can fix this in later PR.